reports/tests.py

```diff
@@ -528,3 +533,96 @@ def _generate_invoice_date(year=datetime.now().year, month=1, this_month=False):
             date = datetime(year, month, 1)
 
         return timezone.make_aware(date)
+
+
+class ReportTaskTestCase(TestCase):
+    """
+    Tests for reports tasks execution
+    """
+
+    def create_user_with_trial_instance(self, key=''):
+        """
+        Returns a trial instance for a user
+
+        :param key: String to append to properties.
+        :return: A trial instance
+        """
+        user, _ = get_user_model().objects.get_or_create(
+            username=key + 'test', email=key + 'test@example.com')
+        instance = OpenEdXInstanceFactory(
+            successfully_provisioned=True, betatestapplication=BetaTestApplication())
+        make_test_appserver(instance=instance, is_active=True)
+        BetaTestApplication.objects.create(
+            user=user,
+            public_contact_email=key + 'publicemail@example.com',
+            subdomain=key + 'betatestdomain',
+            instance_name=instance.name,
+            status=BetaTestApplication.ACCEPTED,
+            instance=instance
+        )
+
+        return instance
+
+    def test_send_trial_instances_report_failure(self):
+        """
+        Tests that if send_trial_instances_report fails because there are no matching instances
+        the recipient gets notified regardless.
+        """
+        send_trial_instances_report(recipients=['to@example.com'])
+        sent_mail = mail.outbox[0]
+        self.assertIn(
+            'Unable to generate a Trial Instances Report due to failure of `instance_statistics_csv`',
+            sent_mail.body
+        )
+        self.assertEqual(
+            [],
+            sent_mail.attachments
+        )
+        self.assertEqual(
+            ['to@example.com'],
+            sent_mail.to
+        )
+        self.assertEqual(len(mail.outbox), 1)
+
+    def test_send_trial_instances_report(self):
```

Comment:
@spokerman12, this test doesn't test the corner cases like the one mentioned [here](https://github.com/open-craft/opencraft/pull/711/files#r580161566).

Reply:
I added the test case as stated in previous comment